context/logger.go

```diff
@@ -39,6 +39,8 @@ type Logger interface {
 	Warn(args ...interface{})
 	Warnf(format string, args ...interface{})
 	Warnln(args ...interface{})
+
+	WithError(err error) *logrus.Entry
```

Comment:
This can use return `Logger`, might as well add `WithField` and `WithFields` as well.

Reply:
Unfortunately, that will cause weird issues, as logrus doesn't implement `WithError(err error) *logrus.Entry`. This is sufficient.